workflow-trace-viewer/src/jvmMain/kotlin/com/squareup/workflow1/traceviewer/App.kt

```diff
@@ -73,16 +93,38 @@ public fun App(
         .align(Alignment.TopEnd)
     )
 
-    // The states are reset when a new file is selected.
-    UploadFile(
-      resetOnFileSelect = {
-        selectedTraceFile = it
-        selectedNode = null
-        frameIndex = 0
-        workflowFrames = emptyList()
+    TraceModeToggleSwitch(
+      onToggle = {
+        resetStates()
+        traceMode = if (traceMode is TraceMode.Live) {
+          frameIndex = 0
+          TraceMode.File(null)
+        } else {
+          // TODO: TraceRecorder needs to be able to take in multiple clients if this is the case
+          /*
+          We set the frame to -1 here since we always increment it during Live mode as the list of
+          frames get populated, so we avoid off by one when indexing into the frames.
+           */
+          frameIndex = -1
+          socket.open()
```

Comment:
I would not open the socket here, since it's not clear who's responsible for closing it.